profile/perf_metics.go

```diff
@@ -0,0 +1,204 @@
+////////////////////////// Backend Bound /////////////////////////
+//////////////////////////////////////////////////////////////////
+func l1Bound(caStallsMemAny, caStallsL1dMiss, clks float64) float64 {
+	// max ( ( CYCLE_ACTIVITY.STALLS_MEM_ANY - CYCLE_ACTIVITY.STALLS_L1D_MISS ) / CLKS , 0 )
+	val := (caStallsMemAny - caStallsL1dMiss) / clks
+	if val < 0 {
+		return 0
+	}
+	return val
+}
+
+func dtlbLoad(dlmStlbHit, dlmWalkDur, dlmWalkCompleted, clks float64) float64 {
+	// ( #Mem_STLB_Hit_Cost * DTLB_LOAD_MISSES.STLB_HIT + DTLB_LOAD_MISSES.WALK_DURATION + 7 * DTLB_LOAD_MISSES.WALK_COMPLETED ) / CLKS
+	return (float64(memSTLBHitCost)*dlmStlbHit + dlmWalkDur + 7*dlmWalkCompleted) / clks
+}
+
+func fbFull(l1pmPending, mlurL1Miss, mlurHitlfb, lpmFbFulls, clks float64) float64 {
+	// Load_Miss_Real_Latency * L1D_PEND_MISS.FB_FULL / CLKS
+	return loadMissRealLatency(l1pmPending, mlurL1Miss, mlurHitlfb) * lpmFbFulls / clks
+}
+
+func l2Bound(caStallsL1dMiss, caStallsL2Miss, clks float64) float64 {
+	// ( CYCLE_ACTIVITY.STALLS_L1D_MISS - CYCLE_ACTIVITY.STALLS_L2_MISS ) / CLKS
+	return (caStallsL1dMiss - caStallsL2Miss) / clks
+}
+
+func l3Bound(mlurL3Hit, mlurL3Miss, caStallsL2Miss, clks float64) float64 {
+	// Mem_L3_Hit_Fraction * CYCLE_ACTIVITY.STALLS_L2_MISS / CLKS
+	return memL3HitFraction(mlurL3Hit, mlurL3Miss) * caStallsL2Miss / clks
+}
+
+func dramBound(mlurL3Hit, mlurL3Miss, caStallsL2Miss, clks float64) float64 {
+	// ( 1 - Mem_L3_Hit_Fraction ) * CYCLE_ACTIVITY.STALLS_L2_MISS / CLKS
+	return (1 - memL3HitFraction(mlurL3Hit, mlurL3Miss)) * caStallsL2Miss / clks
+}
+
+/////////////////////////////////////////////////////////
+////////////////////////// Info /////////////////////////
+/////////////////////////////////////////////////////////
+func slots(clksAny float64) float64 {
+	// #Pipeline_Width * CORE_CLKS
+	return pipelineWidth * coreClock(clksAny)
+}
+
+func coreClock(clksAny float64) float64 {
+	// cpu_clk_unhalted.thread_any / 2
+	return clksAny / 2
+}
+
+func ipc(ins, clks float64) float64 {
+	// inst_retired.any / cpu_clk_unhalted.thread
+	return ins / clks
+}
+
+func loadMissRealLatency(l1pmPending, mlurL1Miss, mlurHitlfb float64) float64 {
+	// l1d_pend_miss.pending / ( mem_load_uops_retired.l1_miss + mem_load_uops_retired.hit_lfb )
+	return l1pmPending / (mlurL1Miss + mlurHitlfb)
+}
+
+func memL3HitFraction(mlurL3Hit, mlurL3Miss float64) float64 {
+	// MEM_LOAD_UOPS_RETIRED.L3_HIT_PS / ( MEM_LOAD_UOPS_RETIRED.L3_HIT_PS + #Mem_L3_Weight * MEM_LOAD_UOPS_RETIRED.L3_MISS_PS )
+	return mlurL3Hit / (mlurL3Hit + float64(memL3Weight)*mlurL3Miss)
+}
+
+///////////////////////////////////////////////////////////////////////////////
+////////////////////////// Auxialiary functions below /////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+func getMetricEventsMap() map[string][]string {
+	return map[string][]string{
+		"CLKS":                   []string{"cpu_clk_unhalted.thread"},
```

Comment:
these must be set in a JSON file, not in the source code.

Reply:
Done